diff --git a/quine.py b/quine.py
index b1720cb..33cfd2e 100755
--- a/quine.py
+++ b/quine.py
@@ -1,15 +1,21 @@
 exec(s:='''
-from matplotlib.pyplot import figure, show
-from PIL import ImageDraw
-from numpy import array, linspace, meshgrid, pi, cos, sin
+from matplotlib.pyplot import (figure,
+show); from PIL.ImageDraw import (Draw
+, Image); from numpy import (cos, sin,
+        pi, linspace, array, meshgrid)
+_B, _E, _D= "\\\\\\\\n", "\\\\n", "3d"
 def f(x):
-    w,h=(len(max(x.split("\\n"),key=len))+1)*6,(x.count("\\n")+1)*15
-    im=ImageDraw.Image.new("L",(w,h))
-    ImageDraw.Draw(im).text((0,0),x,fill=1)
-    THETA,Z=meshgrid(linspace(0,2*pi,w),linspace(0,1,h))
-    X,Y,C=cos(THETA),sin(THETA),[[[i]*3 for i in j] for j in array(im)[::-1]]
-    subplot=figure().add_subplot(projection="3d")
-    subplot.plot_surface(X,Y,Z,rstride=1,cstride=1,facecolors=C)
-    show()
-separated=("exec(s:=\'\'%r\'\')"%s).split("\\\\\\\\n")
-f("\\\\\\\\n".join([i.replace("\\\\n", "\\n") for i in separated]))''')
+  w = len(max(x.split("\\n"),key=len))
+  h,fig=(x.count("\\n")+1)*15,figure()
+  im = Image.new( "L", ( w * 12, h ) )
+  Draw(im).text((0,0), x, fill = True)
+  line = linspace( 0, 2 * pi, w * 12 )
+  T, Z=meshgrid(line, linspace(0,1,h))
+  X,Y,re=cos(T),sin(T),array(im)[::-1]
+  C=[[[i] * 3 for i in j] for j in re]
+  P = fig.add_subplot(projection = _D)
+  P.plot_surface(X, Y, Z, rstride = 1,
+  cstride = 1, facecolors = C); show()
+S= ("exec(s:=\'\'%r\'\')"%s).split(_B)
+f(_B.join([i.replace(_E,"\\n")for i in
+S]))''')
\ No newline at end of file
